cmd/rollapp/drs/upgrade.go

```diff
@@ -0,0 +1,182 @@
+			if err != nil {
+				pterm.Warning.Println("Failed to extract drs version from binary:", err)
+				pterm.Info.Println("Installing the latest version", err)
+			}
+
+			if targetDrs < drsVersion {
+				pterm.Error.Println("Rollapp binary DRS version already at version", drsVersion)
+				return
+			}
+
+			drsInfo, err := firebaseutils.GetLatestDrsVersionCommit(
+				targetDrs,
+				rollerData.HubData.Environment,
+			)
+			if err != nil {
+				pterm.Error.Println("Failed to get the latest commit for target DRS:", err)
+				return
+			}
+
+			var raCommit string
+			switch strings.ToLower(raResp.Rollapp.VmType) {
+			case "evm":
+				raCommit = drsInfo.EvmCommit
+			case "wasm":
+				raCommit = drsInfo.WasmCommit
+			}
+
+			if raCommit == "UNRELEASED" {
+				pterm.Error.Println("rollapp does not support drs version: " + drsVersion)
+				return
+			}
+
+			raNewBinDir, err := os.MkdirTemp(os.TempDir(), "rollapp-drs-update")
+			if err != nil {
+				pterm.Error.Println(
+					"failed to create temporary directory for rollapp binary: ",
+					err,
+				)
+				return
+			}
+			defer os.RemoveAll(raNewBinDir)
+
+			rbi := dependencies.NewRollappBinaryInfo(
+				raResp.Rollapp.GenesisInfo.Bech32Prefix,
+				raCommit,
+				strings.ToLower(raResp.Rollapp.VmType),
+			)
+
+			raDep := dependencies.DefaultRollappDependency(rbi)
+			// override the binary destination with the previously created tmp directory
+			// as the binary will be copied to the final destination after stopping the
+			// system services
+			tmpBinLocation := filepath.Join(
+				raNewBinDir,
+				"rollappd",
+			)
+
+			pterm.Info.Println("starting update")
+			raDep.Binaries[0].BinaryDestination = tmpBinLocation
+			err = dependencies.InstallBinaryFromRepo(raDep, raDep.DependencyName)
+			if err != nil {
+				pterm.Error.Println("failed to install rollapp binary: ", err)
+				return
+			}
+
+			// repalce the current binary with the new one
+			err = archives.MoveBinaryIntoPlaceAndMakeExecutable(
+				tmpBinLocation,
+				consts.Executables.RollappEVM,
+			)
+			if err != nil {
+				pterm.Error.Println("failed to move rollapp binary: ", err)
+				return
+			}
+
+			rollappConfig, err := rollapp.PopulateRollerConfigWithRaMetadataFromChain(
+				home,
+				rollerData.RollappID,
+				rollerData.HubData,
+			)
+
+			damanager := datalayer.NewDAManager(
+				rollappConfig.DA.Backend,
+				rollappConfig.Home,
+				rollappConfig.KeyringBackend,
+			)
+
+			if rollapp.IsDAConfigMigrationRequired(drsVersion, targetDrs, strings.ToLower(raResp.Rollapp.VmType)) {
+				upgradeDaConfig(sequencer.GetDymintFilePath(home), *damanager)
+			}
+
+			err = tomlconfig.UpdateFieldInFile(
+				filepath.Join(home, "roller.toml"),
+				"rollapp_binary_version",
+				raCommit,
+			)
+			if err != nil {
+				pterm.Error.Println("failed to update rollapp binary version in config: ", err)
+				return
+			}
+
+			pterm.Success.Println("update complete")
```

Comment:
this command doesn't stop/start the services, there should be either a request for the user to stop the services or roller should do it for them